src/fsharp/FSComp.txt

```diff
@@ -18,6 +18,8 @@ undefinedNamePatternDiscriminator,"The pattern discriminator '%s' is not defined
 replaceWithSuggestion,"Replace with '%s'"
 missingElseBranch,"The 'if' expression is missing an 'else' branch. The 'then' branch has type '%s'. Because 'if' is an expression, and not a statement, add an 'else' branch which returns a value of the same type."
 elseBranchHasWrongType,"All branches of an 'if' expression must return the same type. This expression was expected to have type '%s' but here has type '%s'."
+followingPatternMatchClauseHasWrongType,"All branches of a pattern match expression must return the same type. This expression was expected to have type '%s' but here has type '%s'."
```

Comment:
Yep. I agree. 
Will also fix the other messages that we already have in there that say "return"